### PR TITLE
chore(build): Disable lombok by default and only enable them in web directories

### PR DESCRIPTION
**Pull request by advancedxy**
### What changes were proposed in this pull request?
1. disable most of the lombok features in the project
2. enable specific lombok features in coordinator and shufffle server's web directories

### Why are the changes needed?
For better code practice
Fix: # (issue)

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No need.

### CODE REVIEW

File: coordinator/src/main/java/org/apache/uniffle/coordinator/web/lombok.config [hunk 1]
```diff
@@ -0,0 +1,4 @@
+clear lombok.allArgsConstructor.flagUsage
+clear lombok.data.flagUsage
+clear lombok.builder.flagUsage
+clear lombok.anyConstructor.flagUsage
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/web/lombok.config:4]: `noArgsConstructor` is also needed

**Comment by jerqi** [comment]
> Per discussion: [#2022 (comment)](https://github.com/apache/incubator-uniffle/pull/2022#discussion_r1708523429), I think we should restrict the usage of lombok. Please take a look at this @jerqi @LuciferYang @xianjingfeng
> 
> Also cc @yl09099 as well.

Ok for me.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/web/lombok.config (hunk 1 above)
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/lombok.config:20]: > Will that be used in the feature?

Yes.  No-argument constructor is required when converting json to an object.

Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/lombok.config]: > So, we haven't converted it to json yet?

 we have converted it to json, but haven't parse it from json.

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Review by LuciferYang — APPROVED** [review]
I strongly support restrictions on the use of Lombok. Let's try to use it as little as possible.

**Pull request merged**